packages/mock/src/msw/index.ts

```diff
@@ -89,7 +83,10 @@ export const generateMSW = (
           ? `export const ${functionName} = () => (${value})\n\n`
           : '',
       handler: `http.${verb}('${route}', async () => {
-        await delay(${getDelay(override)});
+        await delay(${getDelay(
+          override,
+          !isFunction(mock) ? mock : undefined,
+        )});
         return new HttpResponse(${
```

Comment:
The format of the handler response must be different:
`HttpResponse` should be used as static method:
- [json response](https://mswjs.io/docs/api/http-response#httpresponsejsonbody-init)
- [text response](https://mswjs.io/docs/api/http-response#httpresponsetextbody-init)

An example of this would be something like this:
```
return isTextPlan ? HttpResponse.text(`${functionName}()`) : HttpResponse.json(`${functionName}()`);
```

In the msw version, using the HttpResponse class constructor, it returns a type incompatibility bug:
```
Argument of type '() => Promise<HttpResponse>' is not assignable to parameter of type 'ResponseResolver<HttpRequestResolverExtras<PathParams<string>>, DefaultBodyType, undefined>'.
  Type 'Promise<HttpResponse>' is not assignable to type 'AsyncResponseResolverReturnType<undefined>'.
```